src/Mod/Path/PathScripts/PathProfile.py

```diff
@@ -671,12 +671,10 @@ def _processEdges(self, obj):
             # Efor
         # Efor
 
-        delPairs.sort(reverse=True)
-        for p in delPairs:
-            # obj.Base.pop(p)
-            pass
-
-        return shapes
+        if newBase:
+            return shapes, newBase
+        else:
+            return shapes, False
```

Comment:
it's not good practice to return different types - why not return an empty collection `newBase` ?